module/zfs/vdev_raidz.c

```diff
@@ -3450,6 +3537,20 @@ vdev_raidz_get_tsd(spa_t *spa, nvlist_t *nv)
 		 */
 	}
 
+	avl_create(&vdrz->vn_vre.vre_txgs, vedv_raidz_reflow_compare,
+	    sizeof (reflow_node_t), offsetof(reflow_node_t, re_link));
+
+	error = nvlist_lookup_uint64_array(nv, ZPOOL_CONFIG_RAIDZ_EXPAND_TXGS,
+	    &txgs, &txgs_size);
+	if (error == 0) {
+		for (int i = 0; i < txgs_size; i++) {
+			reflow_node_t *re = kmem_zalloc(sizeof (*re), KM_SLEEP);
+			re->re_txg = txgs[i];
+			re->re_logical_width = vdrz->vd_logical_width + i;
+			avl_add(&vdrz->vn_vre.vre_txgs, re);
```

Comment:
Since we now have the EXPANDED_TXGS, I was hoping we could get rid of `ZPOOL_CONFIG_RAIDZ_LOGICAL_WIDTH` (since there is no single logical width), and instead just rely on `ZPOOL_CONFIG_RAIDZ_EXPAND_TXGS`.  We can fill in the AVL by starting at the physical width, and then each entry in EXPANDED_TXGS is for one less.